src/api/procedures/toggleFreezeSto.ts

```diff
@@ -0,0 +1,94 @@
+import BigNumber from 'bignumber.js';
+
+import { PolymeshError, Procedure, SecurityToken, Sto } from '~/internal';
+import { ErrorCode, RoleType, StoStatus, TxTags } from '~/types';
+import { ProcedureAuthorization } from '~/types/internal';
+import { numberToU64, stringToTicker } from '~/utils/conversion';
+
+export interface ToggleFreezeStoParams {
+  id: BigNumber;
+  freeze: boolean;
+}
+
+/**
+ * @hidden
+ */
+export type Params = ToggleFreezeStoParams & {
+  ticker: string;
+};
+
+/**
+ * @hidden
+ */
+export async function prepareToggleFreezeSto(
```

Comment:
Should throw an error if the STO has reached its end date

"The STO has already ended"